src/bin/cargo/cli.rs

```diff
@@ -169,6 +170,17 @@ fn expand_aliases(
                     cmd,
                 ))?;
             }
+            (Some(_), None) => {
+                // Command is built-in and is not conflicting with alias, but contains ignored values.
+                if let Some(mut values) = args.values_of("") {
+                    config.shell().warn(format!(
```

Comment:
should this be error?

Reply:
I think an error would be fine.

Reply:
@ehuss It can break backward compatability

Reply:
If you're concerned about back compat, that's fine too. It can always be changed to an error in the future if desired.